File: C/string_anagram.c
//Anagram Check
#include <stdio.h>
#include <string.h>
#include <ctype.h>

void sortString(char str[]) {
    int len = strlen(str);
    for(int i = 0; i < len-1; i++) {
        for(int j = i+1; j < len; j++) {
            if(str[i] > str[j]) {
                char temp = str[i];
                str[i] = str[j];
                str[j] = temp;
            }
        }
    }
}

int main() {
    char str1[100], str2[100];
    printf("Enter first string: ");
    gets(str1);
    printf("Enter second string: ");
    gets(str2);

    if(strlen(str1) != strlen(str2)) {
        printf("Not an anagram\n");
    } else {
        sortString(str1);
        sortString(str2);
        if(strcmp(str1, str2) == 0) printf("Anagram\n");
        else printf("Not an anagram\n");
    }
    return 0;
}

File: Java/string_anagram.java
//Anagram Check
import java.util.Arrays;
import java.util.Scanner;

public class string_anagram {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter first string: ");
        String str1 = sc.nextLine();
        System.out.print("Enter second string: "); 
        String str2 = sc.nextLine();

        if(str1.length() != str2.length()) {
            System.out.println("Not an anagram");
        } else {
            char[] a1 = str1.toCharArray();
            char[] a2 = str2.toCharArray();
            Arrays.sort(a1);
            Arrays.sort(a2);
            System.out.println(Arrays.equals(a1, a2) ? "Anagram" : "Not an anagram");
        }
    }
}
